APITests/filename_manifest_validate.py

```diff
@@ -0,0 +1,152 @@
+        manifest = manifest.dropna(axis=1,how='all').dropna(axis=0,how="all")
+        manifest.reset_index(drop=False,names='Component',inplace=True)
+
+        manifest = self._generate_errors(manifest)
+        self._save_manifest(manifest)
+
+        return
+
+
+    def _generate_errors(self, manifest: pd.DataFrame) -> pd.DataFrame:
+        # generate 50% of each type of filename error
+        manifest.iloc[::2,1]=manifest.iloc[::2,1].map(lambda x: f"err {x}")
+        manifest.iloc[1::2,2]=manifest.iloc[1::2,2].map(lambda x: x[:-1:])
+
+        return manifest
+
+    def _save_manifest(self, manifest: pd.DataFrame) -> None:
+        self.manifest_path = f"APITests/test_manifests/filename_{self.num_files}_manifest_{uuid.uuid4()}.csv"
+        manifest.to_csv(self.manifest_path,index=False)
+
+        return
+
+
+    def validate_filename_manifest(self):
+        params = self.params
+        params["data_type"] = "MockFilename"
+        params["dataset_scope"] = "syn61682648"
+        params["asset_view"] = self.asset_view_id
+
+        dt_string, time_diff, status_code_dict = asyncio.run(
+            cal_time_api_call(
+                url=self.base_url,
+                params=params,
+                headers=self.headers,
+                concurrent_threads=CONCURRENT_THREADS,
+                manifest_to_send_func=self.cal_run_time.send_manifest_async,
+                file_path_manifest=self.manifest_path,
+            )
+        )
+
+        return format_run_time_result(
+            endpoint_name="model/validate",
+            description=f"Validate a mockFilename manifest from the example data model. The manifest has around {self.num_files} rows with a 100% error rate.",
+            data_schema="Example data schema",
+            num_rows=self.num_files,  # number of rows of the manifest being validated
+            data_type=params["data_type"],  # data type
+            restrict_rules=False,  # Restrict rules is set to False
+            dt_string=dt_string,
+            num_concurrent=CONCURRENT_THREADS,
+            latency=time_diff,
+            status_code_dict=status_code_dict,
+        )
+
+def monitor_filename_manifest_validator() -> Row:
+    schema_url = "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld"
+    project_sizes=[30,300,3000]
+    rows = []
+    for file_num in project_sizes:
+        mv = FilenameManifestValidate(url=schema_url, num_files=file_num)
+        filename_mv = replace(mv, base_url=f"{DEV_REFACTOR_URL}/model/validate")
+        try:
+            filename_mv.create_test_project()
+            filename_mv.generate_manifest()
+            logger.info(f"Monitoring manifest validation for {file_num} files")
+            rows.append(filename_mv.validate_filename_manifest())
+            srt = StoreRuntime()
+            srt.record_run_time_result_synapse(rows=rows)
+        finally:
+            filename_mv.clean_up_resources()
```

Comment:
Could you also add code to clean up test folders and test files that got created locally?